### ISSUE TITLE
Customize loss_scale meet not matched loss mask

### ISSUE THREAD

**Issue by MHX1203** [modelscope/ms-swift#5908]
here is my customize loss_scale class
```
import torch
import torch.nn as nn
from torch.nn import CrossEntropyLoss
from swift.plugin.loss_scale.loss_scale import LossScale, loss_scale_map, calculate_loss_scale
from swift.llm.template.utils import ContextType
# from swift.plugin.loss_scale.loss_scale.utils import calculate_loss_scale
from typing import List, Optional, Tuple

class TrafficLossScale(LossScale):
    loss_scale_config = '/mnt/data/Codes/swift-plugin/traffic_1.json'

    def get_loss_scale(self,
                       context: str,
                       context_type: ContextType,
                       is_last_round: bool,
                       *,
                       query: Optional[str] = None):

        if context_type == ContextType.RESPONSE:
            return calculate_loss_scale(query, context, self.loss_scale_map['response'], self.loss_scale_map['query'])
        return super().get_loss_scale(context, context_type, is_last_round)

loss_scale_map['traffic'] = TrafficLossScale
```

when i do sft use this customize loss_scale, i meet loss_scale not match with loss mask
```
# Use @register_loss_func to decorate your own loss, use --loss_type xxx to train
@register_loss_func(LossType.loss_scale)
def loss_scale_func(outputs, labels, loss_scale=None, num_items_in_batch=None) -> torch.Tensor:
    """Loss func

    Args:
        outputs: The model outputs
        labels: The labels
        loss_scale: The loss scale
        num_items_in_batch: Number of tokens in the labels of gradient accumulation round that are not -100.

    Returns:

    """
    loss, masks = ce_loss_func(outputs, labels)
    if loss_scale is not None:
        shift_scale = loss_scale[..., 1:].to(masks.device)
        # this line error
        shift_scale = shift_scale[masks]
        loss = (shift_scale * loss)
    if num_items_in_batch is None:
        loss = loss.mean()
    else:
        # compat transformers>=4.46
        loss = loss.sum() / num_items_in_batch
    return loss
```

If there are any other place i should change, or some other problems

If anyone can help me, I am grateful;

**Comment by Jintao-Huang** [comment]
Please take a look at the `is_binary` attribute in LossScale.

**Comment by Jintao-Huang** [comment]
Also pay attention to the logits_to_keep attribute in Trainer.

I suggest upgrading ms-swift to the release/3.8 branch.
